llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -10952,22 +10952,73 @@ void TargetLowering::forceExpandWideMUL(SelectionDAG &DAG, const SDLoc &dl,
                                         SDValue &Hi) const {
   EVT VT = LHS.getValueType();
   assert(RHS.getValueType() == VT && "Mismatching operand types");
+  EVT WideVT = EVT::getIntegerVT(*DAG.getContext(), VT.getSizeInBits() * 2);
```

Comment:
scalar size?

Reply:
This is not currently used by vectors and probably needs more fixes to support vector.